src/main/java/chessmaster/game/Game.java

```diff
@@ -66,11 +67,11 @@ public Game(Color playerColour, Color currentTurnColor, ChessBoard board,
         assert (1 <= difficulty) && (difficulty <= 3) : "Difficulty should be between 1 and 3!";
     }
 
-    public void run() {
+    public boolean run() {
```

Comment:
Please help me add a javadoc here and explicitly show what the boolean represents (just chatgpt)

Reply:
Include what the `run` function does as well, something like this:
```java
/**
 * Manages the main gameplay loop of Chess Master.
 *
 * This code segment orchestrates the primary gameplay loop of Chess Master, which encompasses player turns, move handling, and the management of the game state. It starts by displaying the initial game setup, including the chessboard, and then enters a loop where players take turns making moves. If a player enters a valid move, the chessboard is updated, and the game progresses. If an exception is encountered during this process, an error message is displayed to the user. The loop continues until the game ends or specific commands are issued to abort, restart, or exit the game.
 *
 * @return true if the game has ended, either through checkmate or a specific command; false if the game is still ongoing.
 */
```